README.md

```diff
@@ -89,6 +89,22 @@ Please check the [CHANGELOG.md](https://github.com/suvash/hulaaki/blob/master/CH
 
 ## Documentation
```

Comment:
@suvash , I'll start a thread here regarding usage samples. The actual usage of the library remains the same. Since Elixir 1.4 that the applications are inferred from the dependencies and booted with it's supervision tree.

Regarding actual usage, we still didn't include Hulaaki in our project. We will start working on that next week or so.

I've updated the README with more specific transport layer support and configuration, tough.

Reply:
Sounds good. I'll wait for a few weeks until you've been able to integrate this in your project (using your fork), just in case you run into any bugs etc.
Also, I haven't yet gone through the code, but I'm assuming that the Connection Genserver instance is now started by the Application > Client supervision tree. When integrating, can you also check if it's straightforward to open multiple connections (to multiple servers) with this Supervision tree structure ? 
After all looks good, we should be able to clean up and merge.

Reply:
Hey, first off all thank you @suvash for this great library! If you need to connect to multiple servers, you will need to have a client for each server, since the relation of the clients is 1:1 to connections (as it was before if I'm not wrong). The only guarantee that we will need is to have a different `client_id` for each client, which we use as id for each supervised connection.

Reply:
Yeah. I think the Client : Connection is 1:1 now and was before as well. I was thinking more about the Supervision tree that's started as a part of the application. If I've understood this correctly, the application supervision tree will start an instance of the Hulaaki tree (with client and connection genservers), so was thinking if that'd be straightforward to scale to multiple clients+connections as well.
However, I still haven't had the time to look at the code yet. Vacation around the corner and been busy. If I don't get the time this week, I'll most likely only be able to look at towards the end of August.
But, that should be alright, and perhaps enough time for you to integrate it in your Elixir project and test run in production(?). Looking forward to the integration news.
And yeah, thanks a lot for this PR @brunombsilva @ejscunha, Lots of good stuff here that I've been planning to implement for a while but hadn't had the time to. :beers:

Reply:
So, how has it gone with the integration ?

Reply:
Started working on it just now :)
When we get something working we will give news.

Reply:
Hello again, sorry for all this time without saying anything. We started using Hulaaki with websockets in our QA environment to make tests, it looks to be working ok. Sometimes the connection goes down for some reason, maybe due to some issue with the keep alive option but I still need to investigate this further.

In our use case we need to open many connections to a MQTT server, since we can't re use the same connection for different topics, so while trying to include the clients in our supervision tree it doesn't feel natural and I think it needs to be improved.

We setup a DynamicSupervisor responsible for supervising all the new clients on demand and to start the clients we have a task that requests the supervisor to start each client.

Something like this:

``` elixir
```

Reply:
Hello again, sorry for all this time without saying anything. We started using Hulaaki with websockets in our QA environment to make tests, it looks to be working ok. Sometimes the connection goes down for some reason, maybe due to some issue with the keep alive option but I still need to investigate this further.

In our use case we need to open many connections to a MQTT server, since we can't re use the same connection for different topics, so while trying to include the clients in our supervision tree it doesn't feel natural and I think it needs to be improved.

We setup a DynamicSupervisor responsible for supervising all the new clients on demand and to start the clients we have a task that requests the supervisor to start each client.

Something like this:

``` elixir
defmodule Bootstrapper do
  use Task
  alias Clients.Supervisor

  def start_link(_) do
    Task.start_link(__MODULE__, :run, [])
  end

  def run do
      clients = [..]

      Enum.each(clients, fn client ->
        Supervisor.start_client(client)
      end)
  end
end

defmodule Clients.Supervisor do
  use DynamicSupervisor

  def start_link(_) do
    DynamicSupervisor.start_link(__MODULE__, :ok, name: __MODULE__)
  end

  def init(:ok) do
    DynamicSupervisor.init(strategy: :one_for_one)
  end

  def start_client(client) do
    child_spec = %{
      id: client,
      start: {Client, :start_link, [:start, client]},
      restart: :transient
    }

    DynamicSupervisor.start_child(__MODULE__, child_spec)
  end
end

defmodule Client do
  use Hulaaki.Client

  def start_link(:start, client) do
    case __MODULE__.start_link(%{}) do
      {:ok, pid} ->
        send(pid(, {:connect_and_subscribe, client})
        {:ok, pid}

      other ->
        other
    end
  end

  def handle_info({:connect_and_subscribe, client}, state) do
    Task.start(__MODULE__, :connect_and_subscribe, [client, self()])

    {:noreply, state}
  end

  def connect_and_subscribe(client, pid) do
    with {:ok, topic} <- connect({client, pid}),
            :ok <- subscribe(pid, topics: [topic], qoses: [1]) do
      :ok
    else
      error -> error
    end
  end

  defp connect({client, pid}) do
    ..
    opts = [..]

    case __MODULE__.connect(pid, opts) do
     end
  end
end
```

Reply:
Hi again @ejscunha , Thanks for the feedback above. I was sorta expecting issues around the supervisor, it's kinda tricky to setup one type of tree that might work for all end users of the library.
I'm planning to incorporate these changes into the library sometime in October. Do you have any new insights/feedback that I could make use of, based on any further changes you've had to do ?